lib/pedicel/base.rb

```diff
@@ -142,15 +136,20 @@ def verify_signature(ca_certificate_pem: Pedicel.config[:apple_root_ca_g3_cert_p
 
     private
 
-    def self.verify_signature_certificate_oids(signature:)
-      leaf = signature.certificates.find{|c| c.extensions.find{|e| e.oid == Pedicel.config[:oids][:leaf_certificate]}}
+    def self.verify_signature_certificate_oids(signature:, config: Pedicel.config)
+      leaf = signature.certificates.find do |c|
+        c.extensions.find do |e|
+          e.oid == config[:oids][:leaf_certificate]
+        end
+      end
```

Comment:
If expanded, then `c` -> `certificate` and `e` -> `extension`.

And if so, likewise below where `intermediate` is extracted.

Or something like ...

```ruby
leaf, intermediate = nil, nil

signature.certificates.each do |certificate|
  certificate.extensions.each do |extension|
    case extension.oid
    when config[:oids][:leaf_certificate]
      leaf = certificate
    when config[:oids][:intermediate_certificate]
      intermediate = certificate
    end
  end
  break if leaf && intermediate
end
```

Reply:
Good idea!

Reply:
Uhm, if we are "complexifying" this anyway, then ... why not run through all certificates (thus, no `break`) and err also if the leaf/intermediate OID is met more than once?